### DISCUSSION TITLE
Is disabling line wrapping altogether possible?

### DISCUSSION THREAD

**Discussion by elidonner** [astral-sh/ruff#15096] [Q&A] [answered]
I'd like to be able to disable line wrapping altogether. I do not just want to set the line length to the max of 320.

I've tried the following
```
[tool.ruff.lint]
ignore = ["E501"]
unfixable = ["E501"]
```

This doesn't seem to work. Anyone have success with this? I like using other features, like fixing spacing with my comments, consistent quotations, mccabe checking, but I really would love to disable line wrapping.

**Comment by MichaReiser** [comment]
Hi. Can you tell me a bit more about what commands you're using? Do you use `ruff check` or `ruff format` or both? Do you have an example where you want to disable line wrapping?

**Comment by elidonner** [comment]
I use `ruff format`

Here's an example, wrapped:
![image](https://github.com/user-attachments/assets/5c3961d0-84fd-4782-9364-2f4880509a8c)

Unwrapped:
![image](https://github.com/user-attachments/assets/2ca990e0-1dae-4a27-bf94-b0c13d7b6e92)

There's examples like this all over the codebase I'm referencing.
The general idea being, it's a small codebase where maintainers can be specific about their choice of line wrapping. However, it would still be nice to catch improper spaces on comments, quotation consistency, etc.

**Reply by MichaReiser** [reply, marked as answer]
Thanks. The example is very helpful.

No, the formatter has no such option. You could try some of the [pycodestyle](https://docs.astral.sh/ruff/rules/#error-e) `E2XX` rules to see if they can replicate your desired behavior, but many of them, unfortunately, have no autofix today. So they might not be very useful to you.

**Reply by elidonner** [reply]
Thanks for the definitive response. With the understanding that one of the main points of the formatter is line wrapping... it would still be cool if there were an option to disable this part of the tool
